File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
*/

class Solution {
public:
    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> answer;
        postOrder(root,answer);
        return answer;
    }

    void postOrder(TreeNode* recievedRoot, vector<int> &answer){
        if (recievedRoot == nullptr) return;
        postOrder(recievedRoot->left,answer);
        postOrder(recievedRoot->right,answer);
        answer.push_back(recievedRoot->val);
    }
};

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        array = []
        self.traverse(root,array)
        return array

    def traverse(self,root,array):
        
        if root == None:
            return
        
        self.traverse(root.left,array)
        self.traverse(root.right,array)
        array.append(root.val)



        
        